File: plataformas-danos-back/src/main/java/com/plataformas_danos_back/service/CotizacionService.java
package com.plataformas_danos_back.service;

import com.plataformas_danos_back.model.dto.CotizacionRequest;
import com.plataformas_danos_back.model.dto.CotizacionResponse;

public interface CotizacionService {

    /**
     * Inicia una nueva cotización obteniendo un folio único del Servicio de Folios
     * (Plataforma-core-ohs). La cotización se crea con estadoValidacion=INCOMPLETA.
     *
     * @return cotización recién creada con folio asignado
     */
    CotizacionResponse iniciar();

    /**
     * Obtiene los datos de una cotización por su folio de negocio.
     *
     * @param folio folio en formato COT-YYYY-NNNNNN
     * @return cotización encontrada
     */
    CotizacionResponse obtenerPorFolio(String folio);

    /**
     * Actualiza los datos generales de una cotización existente.
     * Aplica validación de RFC, rango de fechas, IDs de catálogos y versionado optimista.
     *
     * @param folio   folio en formato COT-YYYY-NNNNNN
     * @param request datos a actualizar (todos opcionales excepto version)
     * @return cotización actualizada
     */
    CotizacionResponse actualizar(String folio, CotizacionRequest request);
}

File: cotizador-danos-web/src/lib/services/cotizacionService.ts
import axios from 'axios';
import type { CotizacionResponse, CotizacionRequest } from '@/lib/schemas/cotizacion.schema';

const api = axios.create({
  baseURL: process.env.NEXT_PUBLIC_API_URL,
});

/**
 * Inicia una nueva cotización.
 * POST /api/v1/cotizaciones
 *
 * @returns La cotización creada con folio asignado y estado INCOMPLETA.
 */
export async function iniciarCotizacion(token: string): Promise<CotizacionResponse> {
  const response = await api.post<CotizacionResponse>(
    '/api/v1/cotizaciones',
    {},
    {
      headers: { Authorization: `Bearer ${token}` },
    }
  );
  return response.data;
}

/**
 * Obtiene los datos generales de una cotización por su folio.
 * GET /api/v1/cotizaciones/{folio}
 */
export async function getCotizacion(folio: string, token: string): Promise<CotizacionResponse> {
  const response = await api.get<CotizacionResponse>(
    `/api/v1/cotizaciones/${encodeURIComponent(folio)}`,
    {
      headers: { Authorization: `Bearer ${token}` },
    }
  );
  return response.data;
}

/**
 * Actualiza los datos generales de una cotización existente.
 * PUT /api/v1/cotizaciones/{folio}
 *
 * Incluye el campo version para versionado optimista (HTTP 409 si hay conflicto).
 */
export async function actualizarCotizacion(
  folio: string,
  data: CotizacionRequest,
  token: string
): Promise<CotizacionResponse> {
  const response = await api.put<CotizacionResponse>(
    `/api/v1/cotizaciones/${encodeURIComponent(folio)}`,
    data,
    {
      headers: { Authorization: `Bearer ${token}` },
    }
  );
  return response.data;
}
